Source/modules/webcl/WebCL.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2011 Samsung Electronics Corporation. All rights reserved.
+// Copyright (C) 2015 Intel Corporation All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WebCL_h
+#define WebCL_h
+
+#if ENABLE(WEBCL)
+#include "core/html/HTMLCanvasElement.h"
+#include "core/html/HTMLImageElement.h"
+#include "core/html/HTMLVideoElement.h"
+#include "core/html/ImageData.h"
+#include "modules/webcl/WebCLCallback.h"
+#include "modules/webcl/WebCLCommandQueue.h"
+#include "modules/webcl/WebCLConfig.h"
+#include "modules/webcl/WebCLContext.h"
+#include "modules/webcl/WebCLDevice.h"
+#include "modules/webcl/WebCLExtension.h"
+#include "modules/webcl/WebCLEvent.h"
+#include "modules/webcl/WebCLImage.h"
+#include "modules/webcl/WebCLInputChecker.h"
+#include "modules/webcl/WebCLKernel.h"
+#include "modules/webcl/WebCLMemoryObject.h"
+#include "modules/webcl/WebCLOpenCL.h"
+#include "modules/webcl/WebCLPlatform.h"
+#include "modules/webcl/WebCLProgram.h"
+#include "modules/webcl/WebCLSampler.h"
+
+#include <stdlib.h>
+#include <wtf/OwnPtr.h>
+#include <wtf/PassRefPtr.h>
+#include <wtf/RefCounted.h>
+#include <wtf/RefPtr.h>
+#include <wtf/Threading.h>
+
+using namespace std;
+
+namespace blink {
+
+typedef unsigned CLenum;
+class ExceptionState;
+class ExecutionContext;
+class ImageData;
+class ImageBuffer;
+class IntSize;
+class WebCLPlatform;
+
+class WebCL : public ThreadSafeRefCounted<WebCL>, public ScriptWrappableBase {
```

Comment:
why it inherit ThreadSafeRefCounted? you don't ref or deref it in other thread.
I understand why you want to use ThreadSafeRefCounted, so change the code to ref/deref in other thread.